presto-main/src/main/java/com/facebook/presto/operator/aggregation/noisyaggregation/sketch/Bitmap.java

```diff
@@ -0,0 +1,194 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.operator.aggregation.noisyaggregation.sketch;
+
+import com.google.common.annotations.VisibleForTesting;
+import io.airlift.slice.SizeOf;
+import io.airlift.slice.SliceInput;
+import org.openjdk.jol.info.ClassLayout;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+/**
+ * A level of abstraction over the bitmaps used in sketches such as LPCA and SFM.
+ * These are essentially arrays of booleans that support flipping and applying randomized response.
+ * Concretely, these are stored as byte arrays.
+ */
+public class Bitmap
```

Comment:
Have you considered https://github.com/RoaringBitmap/RoaringBitmap?

Reply:
I hadn't! Thanks for the link. In this case, I'm not sure a compressed bitmap will offer much value here, since in the end we scatter random bits throughout the bitmap. It'd be worth confirming in the future, but for now, I think we can just keep the uncompressed bitmap.

That said, the `Bitmap` class should probably extend Java's `BitSet`.

Reply:
As mentioned above, the `Bitmap` class now wraps `BitSet`.